src/test/scala/com/fulcrumgenomics/vcf/api/VcfIoTest.scala

```diff
@@ -227,12 +229,51 @@ class VcfIoTest extends UnitSpec with OptionValues {
     v.gt("0").callIndices.mkString("/") shouldBe "2/3"
   }
 
-  it should "not attempt to index a VCF when streaming to a file handle or other kind of non-regular file" in {
+  it should "not allow a VCF writer to write to a directory path" in {
     val samples = Seq("sample")
-    val builder = VcfBuilder(samples=samples)
-    val writer  = VcfWriter(Io.DevNull, header = builder.header)
-    builder.add(chrom = "chr1", pos = 100, alleles = Seq("A", "C"), gts = Seq(Gt(sample="sample", gt="0/1")))
-    noException shouldBe thrownBy { writer.write(builder.toSeq) }
+    val builder = VcfBuilder(samples = samples)
+    val output  = Io.makeTempDir(getClass.getSimpleName)
+    output.toFile.deleteOnExit()
+    an[IllegalArgumentException] shouldBe thrownBy { VcfWriter(output, header = builder.header) }
+  }
+
+  it should "write a sibling index when writing to a plaintext VCF file" in {
+    val samples = Seq("sample")
+    val builder = VcfBuilder(samples = samples)
+    val output  = makeTempFile(getClass.getSimpleName, VcfExtension)
+    val writer  = VcfWriter(output, header = builder.header)
+    builder.add(chrom = "chr1", pos = 100, alleles = Seq("A", "C"), gts = Seq(Gt(sample = "sample", gt = "0/1")))
     writer.close()
+    Files.exists(PathUtil.replaceExtension(output, VcfExtension + IdxExtension)) shouldBe true
+  }
+
+  it should "write a sibling index when writing to a compressed VCF file" in {
+    val samples = Seq("sample")
+    val builder = VcfBuilder(samples = samples)
+    val output  = makeTempFile(getClass.getSimpleName, VcfGzExtension)
+    val writer  = VcfWriter(output, header = builder.header)
+    builder.add(chrom = "chr1", pos = 100, alleles = Seq("A", "C"), gts = Seq(Gt(sample = "sample", gt = "0/1")))
+    writer.close()
+    Files.exists(PathUtil.replaceExtension(PathUtil.removeExtension(output), VcfGzExtension + TbiExtension)) shouldBe true
```

Comment:
Consider using `VcfSource(output).isQueryable shouldBe true` as this will only return true if the index exists, but will handle things better if the default index format changes (e.g. to CSI).

Reply:
Smart!